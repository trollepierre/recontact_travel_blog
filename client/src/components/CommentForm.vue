<template>
  <form
    class="forum__form"
    @submit="submitComment">
    <p class="form-title">{{ $t("addComment") }}</p>
    <div class="input-group author">
      <label
        class="forum__label"
        for="author">
        {{ $t("name") }}
      </label>
      <input
        id="author"
        v-model="newAuthor"
        class="forum__comment-box author"
        :placeholder="anonymous">
    </div>
    <div class="input-group comment">
      <label
        class="forum__label"
        for="comment">
        {{ $t("text") }}
      </label>
      <input
        id="comment"
        v-model="newComment"
        class="forum__comment-box text"
        :placeholder="textPlaceholder">
    </div>
    <button
      type="submit"
      class="form-button">
      {{ $t("send") }}
    </button>
  </form>
</template>
<script>
  import commentsApi from '../api/comments'
  import notificationsService from '../services/notifications'

  export default {
    name: 'CommentForm',
    data() {
      return {
        newComment: '',
        newAuthor: '',
        errorComment: '',
        anonymous: this.$t('anonymous'),
        textPlaceholder: this.$t('textPlaceholder'),
        dropboxId: parseInt(this.$route.params.id, 10),
      }
    },
    methods: {
      submitComment(e) {
        e.preventDefault()
        if (this.newComment !== '') {
          const comment = {
            text: this.newComment,
            author: this.newAuthor,
          }
          return commentsApi.send(this.dropboxId, comment)
            .then(this.displaySuccessNotification)
            .catch(() => {
              this.displayErrorNotification()
            })
        }
      },
      displaySuccessNotification() {
        notificationsService.success(this, this.$t('commentSuccess'))
      },
      displayErrorNotification() {
        notificationsService.error(this, this.$t('commentError'))
      },
    },
    i18n: {
      messages: {
        fr: {
          addComment: 'Ajouter un commentaire',
          name: 'De la part de',
          commentError: 'Erreur lors de la prise en compte de ton commentaire.',
          commentSuccess: 'Ton commentaire a été pris en compte.',
          anonymous: 'Anonyme',
          textPlaceholder: 'N\'hésitez pas à ajouter des commentaires !',
          send: 'Envoyer',
          text: 'Votre message',
        },
        en: {
          addComment: 'Add a comment',
          name: 'From',
          commentError: 'Error when adding the comment.',
          commentSuccess: 'Your comment has been taken into consideration.',
          anonymous: 'Anonymous',
          textPlaceholder: 'Add your comment here!',
          send: 'Send',
          text: 'Your message',
        },
      },
    },
  }
</script>

<style>
  .forum__form {
    margin: 20px;
  }

  .form-title {
    font-size: 1.5em;
    font-weight: bold;
  }

  .input-group {
    display: flex;
    justify-content: center;
    flex-direction: column;
    margin-top: 10px;
    margin-bottom: 20px;
  }

  .forum__label {
    font-size: 1.5em;
    font-weight: bold;
    margin-bottom: 10px;
  }

  .forum__comment-box {
    font-size: 1.5em;
    align-items: start;
    border-radius: 3px;
    border: 1px solid rgba(255, 255, 255, 0.7);
    padding-left: 10px;
  }

  .forum__comment-box.text {
    height: 100px;
  }

  button.form-button {
    text-transform: uppercase;
    color: #d14800;
    background: #ffffff;
    border: 1px solid #d14800;
    cursor: pointer;
    padding: 15px 30px;
    border-radius: 4px;
    width: 100%;
    margin-bottom: 10px;
    font-weight: 700;
    max-width: 300px;
  }
</style>
