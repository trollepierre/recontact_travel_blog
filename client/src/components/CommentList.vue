<template>
  <div
    v-if="!isEmptyPlus(comments)"
    class="comments-list">
    <h2 class="forum__name">
      {{ $t("hereTheComments") }}
    </h2>
    <ul class="forum__comment-list">
      <li
        v-for="comment in comments"
        :key="comment.text"
        class="comment__item">
        <comment-card :comment="comment"/>
      </li>
    </ul>
  </div>
</template>
<script>
  import CommentCard from './common/CommentCard.vue'
  import commentsApi from '../api/comments'
  import { isEmptyPlus } from '../../../server/src/domain/utils/ramda-utils'

  export default {
    name: 'CommentList',
    components: { CommentCard },
    data() {
      return {
        comments: [],
      }
    },
    mounted() {
      this.fetchComments()
    },
    methods: {
      isEmptyPlus,
      fetchComments() {
        commentsApi.fetch(this.dropboxId)
          .then(comments => {
            this.comments = comments
          })
      },
    },
    i18n: {
      messages: {
        fr: {
          hereTheComments: 'Voici les commentaires de l\'article !',
        },
        en: {
          hereTheComments: 'Here the comments of the article!',
        },
      },
    },
  }
</script>
