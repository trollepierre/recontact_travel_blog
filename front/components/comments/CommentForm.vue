<template>
  <div>
    <h2 class="form__title">
      {{ $t("addComment") }}
    </h2>
    <form
      class="form"
      @submit="submitComment">
      <header class="form__header">
        <div class="input-group author">
          <label
            class="form__label author"
            for="author">
            {{ $t("name") }}
          </label>
          <input
            id="author"
            v-model="newAuthor"
            :placeholder="anonymous"
            autocomplete="off"
            spellcheck="false"
            class="forum__comment-box author">
        </div>
      </header>
      <div class="form__content">
        <div class="input-group comment">
          <label
            class="form__label comment"
            for="comment">
            {{ $t("text") }}
          </label>
          <textarea
            id="comment"
            v-model="newComment"
            :placeholder="textPlaceholder"
            class="forum__comment-box text"/>
        </div>
      </div>
      <footer class="form__footer">
        <app-button
          type="submit"
          class="form-button"
          :text="$t('send')"
          @click="submitComment"/>
      </footer>
    </form>
  </div>
</template>
<script>
  import AppButton from '@/components/AppButton/AppButton'
  import { isEmptyPlus as isEmpty } from '../../services'
  import commentsApi from '../../services/api/comments'
  import notificationsService from '../../services/services/notifications'

  export default {
    name: 'CommentForm',
    components: { AppButton },
    data() {
      return {
        newComment: '',
        newAuthor: '',
        errorComment: '',
        anonymous: this.$t('yourName'),
        textPlaceholder: this.$t('textPlaceholder'),
        dropboxId: parseInt(this.$route.params.id, 10),
      }
    },
    methods: {
      submitComment() {
        if (this.newComment !== '') {
          const comment = {
            text: this.newComment,
            author: this.newAuthor,
          }
          return commentsApi.send(this.dropboxId, comment)
            .then(this.resetNewComment)
            .then(this.reloadComments)
            .then(this.displaySuccessNotification)
            .catch(this.displayErrorNotification)
        }
        return true
      },
      reloadComments() {
        this.$emit('reload')
      },
      resetNewComment() {
        this.newComment = ''
      },
      displaySuccessNotification() {
        notificationsService.information(this.$t('commentSuccess'))
      },
      displayErrorNotification() {
        notificationsService.error(this.$t('commentError'))
      },
      isEmpty,
    },
    i18n: {
      messages: {
        fr: {
          addComment: 'Ajouter un commentaire',
          name: 'De la part de',
          yourName: 'Ton nom',
          commentError: 'Erreur lors de la prise en compte de ton commentaire.',
          commentSuccess: 'Ton commentaire a été pris en compte.',
          anonymous: 'Anonyme',
          textPlaceholder: 'N\'hésitez pas à ajouter des commentaires !',
          send: 'Envoyer',
          text: 'Votre message',
        },
        en: {
          addComment: 'Add a comment',
          name: 'From',
          yourName: 'Your name',
          commentError: 'Error when adding the comment.',
          commentSuccess: 'Your comment has been taken into consideration.',
          anonymous: 'Anonymous',
          textPlaceholder: 'Add your comment here!',
          send: 'Send',
          text: 'Your message',
        },
      },
    },
  }
</script>

<style lang="scss" scoped>
  .form__title {
    font-size: 1.5em;
    font-weight: bold;
    font-family: serif;
  }

  .form {
    min-width: 260px;
    background: $comment-form-bg;
    border-radius: 4px;
    border: 1px solid $border;
    display: flex;
    flex-direction: column;
    width: 50%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 30px;
  }

  .form__header, .form__content {
    padding: 15px;
    height: auto;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .forum__comment-box {
    font-family: serif;
    font-size: 16px;
    font-weight: 400;
    margin: 0;
    overflow-wrap: break-word;
    padding-left: 10px;
    border: 1px solid $border;
    background-color: $comment-form-input-bg;
    border-radius: 4px;
    color: $comment-form-color;
  }

  .input-group {
    display: flex;
    justify-content: center;
    flex-direction: column;
    margin-bottom: 5px;
    width: 80%;
  }

  .input-group.author {
    width: 60%;
    max-width: 300px;
  }

  .form__label {
    font-size: 16px;
    font-weight: 700;
    font-family: serif;
    color: $comment-form-label;
    margin: 0;
    overflow-wrap: break-word;
    padding-bottom: 10px;
  }

  .forum__comment-box.text {
    font-weight: 400;
    height: 100px;
    word-break: keep-all;
    padding: 10px;
  }

  .form__footer {
    text-align: center;
    padding: 15px;
    border-top: 1px solid $border;
  }

  .form-button {
    max-width: 300px;
  }

  @media only screen and (max-width: 1000px) {
    .form {
      width: 100%;
    }
  }
</style>
