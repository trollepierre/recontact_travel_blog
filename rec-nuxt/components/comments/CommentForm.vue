<template>
  <div class="comment__form">
    <p class="form__title">
      {{ $t("addComment") }}
    </p>
    <form
      class="form"
      @submit="submitComment">
      <header class="form__header">
        <div class="input-group author">
          <label
            v-if="!isEmpty(newAuthor)"
            class="form__label author"
            for="author">
            {{ $t("name") }}
          </label>
          <input
            id="author"
            v-model="newAuthor"
            :placeholder="anonymous"
            class="forum__comment-box author">
        </div>
      </header>
      <main class="form__content">
        <div class="input-group comment">
          <label
            class="form__label comment"
            for="comment">
            {{ $t("text") }}
          </label>
          <textarea
            id="comment"
            v-model="newComment"
            :placeholder="textPlaceholder"
            class="forum__comment-box text"/>
        </div>
      </main>
      <footer class="form__footer">
        <button
          type="submit"
          class="form-button">
          {{ $t("send") }}
        </button>
      </footer>
    </form>
  </div>
</template>
<script>
  import { isEmpty } from 'ramda'
  import commentsApi from '../../services/api/comments'
  import notificationsService from '../../services/services/notifications'

  export default {
    name: 'CommentForm',
    data() {
      return {
        newComment: '',
        newAuthor: '',
        errorComment: '',
        anonymous: this.$t('name'),
        textPlaceholder: this.$t('textPlaceholder'),
        dropboxId: parseInt(this.$route.params.id, 10),
      }
    },
    methods: {
      submitComment(e) {
        e.preventDefault()
        if (this.newComment !== '') {
          const comment = {
            text: this.newComment,
            author: this.newAuthor,
          }
          return commentsApi.send(this.dropboxId, comment)
            .then(this.resetNewComment)
            .then(this.reloadComments)
            .then(this.displaySuccessNotification)
            .catch(this.displayErrorNotification)
        }
      },
      reloadComments() {
        this.$emit('reload')
      },
      resetNewComment() {
        this.newComment = ''
      },
      displaySuccessNotification() {
        notificationsService.success(this, this.$t('commentSuccess'))
      },
      displayErrorNotification() {
        notificationsService.error(this, this.$t('commentError'))
      },
      isEmpty,
    },
    i18n: {
      messages: {
        fr: {
          addComment: 'Ajouter un commentaire',
          name: 'De la part de',
          commentError: 'Erreur lors de la prise en compte de ton commentaire.',
          commentSuccess: 'Ton commentaire a été pris en compte.',
          anonymous: 'Anonyme',
          textPlaceholder: 'N\'hésitez pas à ajouter des commentaires  !',
          send: 'Envoyer',
          text: 'Votre message',
        },
        en: {
          addComment: 'Add a comment',
          name: 'From',
          commentError: 'Error when adding the comment.',
          commentSuccess: 'Your comment has been taken into consideration.',
          anonymous: 'Anonymous',
          textPlaceholder: 'Add your comment here!',
          send: 'Send',
          text: 'Your message',
        },
      },
    },
  }
</script>

<style scoped>
  .form__title {
    font-size: 1.5em;
    font-weight: bold;
  }

  .form {
    min-width: 260px;
    background: #ffffff;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px rgba(0, 0, 0, .15);
    border: 1px solid rgba(0, 0, 0, .09);
    display: flex;
    flex-direction: column;
    color: #535a60;
    width: 50%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 30px;
  }

  .form__header, .form__content {
    border-bottom: 1px solid #e6e6e6;
    padding: 15px;
    height: auto;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .forum__comment-box {
    font-size: 16px;
    font-weight: 400;
    margin: 0;
    overflow-wrap: break-word;
    padding-left: 10px;
    border-width: 1px;
    border-style: solid;
    border-color: gray;
    border-image: initial;
  }

  .input-group {
    display: flex;
    justify-content: center;
    flex-direction: column;
    margin-bottom: 5px;
    width: 80%;
  }

  .input-group.author {
    width: 60%;
    max-width: 300px;
  }

  .form__label {
    font-size: 16px;
    font-weight: 700;
    color: #07c;
    margin: 0;
    overflow-wrap: break-word;
    padding-bottom: 10px;
  }

  .forum__comment-box.text {
    font-weight: 400;
    height: 100px;
    word-break: keep-all;
    padding: 10px;
  }

  .form__footer {
    text-align: center;
    padding: 15px;
    border-top: 1px solid #e6e6e6;
  }

  button.form-button {
    text-transform: uppercase;
    color: #d14800;
    background: #ffffff;
    border: 1px solid #d14800;
    cursor: pointer;
    padding: 15px 30px;
    border-radius: 4px;
    width: 100%;
    margin-bottom: 10px;
    font-weight: 700;
    max-width: 300px;
  }

  @media only screen and (max-width: 1000px) {
    .form {
      width: 100%;
    }
  }
</style>
