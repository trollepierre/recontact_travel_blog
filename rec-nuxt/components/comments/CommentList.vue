<template>
  <div
    v-if="!isEmpty(comments)"
    class="comments-list">
    <h2 class="forum__name">
      {{ $t("hereTheComments") }}
    </h2>
    <ul class="forum__comment-list">
      <li
        v-for="comment in comments"
        :key="comment.createdAt"
        class="comment__item">
        <comment-card :comment="comment"/>
      </li>
    </ul>
  </div>
</template>

<script>
  import { isEmpty } from 'ramda'
  import CommentCard from '../common/CommentCard.vue'
  import commentsApi from '../../services/api/comments'

  export default {
    name: 'CommentList',
    components: { CommentCard },
    props: {
      toReload: { type: Boolean, default: () => false },
    },
    data() {
      return {
        comments: [],
        dropboxId: parseInt(this.$route.params.id, 10),
      }
    },
    watch: {
      async toReload() {
        await commentsApi.fetch(this.dropboxId)
          .then(comments => {
            this.comments = comments
          })
          .then(this.reloaded)
      },
    },
    mounted() {
      commentsApi.fetch(this.dropboxId)
        .then(comments => {
          this.comments = comments
        })
    },
    methods: {
      isEmpty,
      reloaded() {
        this.$emit('reloaded')
      },
    },
    i18n: {
      messages: {
        fr: {
          hereTheComments: 'Voici les commentaires de l\'article  !',
        },
        en: {
          hereTheComments: 'Here the comments of the article!',
        },
      },
    },
  }
</script>

<style scoped>
  .forum__comment-list {
    padding: 0;
    text-align: center;
    justify-content: center;
    align-items: center;
  }

  .comment__item {
    display: block;
    margin-bottom: 20px;
    width: 50%;
    margin-left: auto;
    margin-right: auto;
  }

  @media only screen and (max-width: 1000px) {
    .comment__item {
      width: 100%;
    }
  }
</style>
