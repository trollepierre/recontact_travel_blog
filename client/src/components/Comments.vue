<template>
  <div class="comments">
    <comment-form
      :new-comment="newComment"
      :submit-comment="submitComment"/>
    <comment-list :comments="comments"/>
  </div>
</template>
<script>
  import CommentForm from './CommentForm.vue'
  import CommentList from './CommentList.vue'
  import notificationsService from '../services/notifications'
  import commentsApi from '../api/comments'

  export default {
    name: 'Comments',
    components: { CommentForm, CommentList },
    data() {
      return {
        comments: [],
        newComment: '',
        errorComment: '',
      }
    },
    mounted() {
      this.fetchComments()
    },
    methods: {
      fetchComments() {
        commentsApi.fetch(this.dropboxId)
          .then(comments => {
            this.comments = comments
          })
      },
      submitComment(e) {
        e.preventDefault()
        if (this.newComment !== '') {
          return commentsApi.send(this.dropboxId, this.newComment)
            .then(this.displaySuccessNotification)
            .catch(() => {
              this.displayErrorNotification()
            })
        }
      },
      displaySuccessNotification() {
        notificationsService.success(this, this.$t('commentSuccess'))
      },
      displayErrorNotification() {
        notificationsService.error(this, this.$t('commentError'))
      },
    },
    i18n: {
      messages: {
        fr: {
          addComment: 'Ajouter un commentaire',
          commentError: 'Erreur lors de la prise en compte de ton commentaire.',
          commentSuccess: 'Ton commentaire a été pris en compte.',
        },
        en: {
          addComment: 'Add a comment',
          commentError: 'Error when adding the comment.',
          commentSuccess: 'Your comment has been taken into consideration.',
        },
      },
    },
  }
</script>
