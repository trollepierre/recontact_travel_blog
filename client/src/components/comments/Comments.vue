<template>
  <div class="comments">
    <comment-list
      :to-reload="toReload"
      @reloaded="reloadOff"/>
    <comment-form @reload="reloadTo"/>
  </div>
</template>
<script>
  import CommentForm from './CommentForm.vue'
  import CommentList from './CommentList.vue'

  export default {
    name: 'Comments',
    components: { CommentForm, CommentList },
    data() {
      return {
        toReload: false,
      }
    },
    methods: {
      // TODO use store
      reloadTo() {
        this.toReload = true
      },
      reloadOff() {
        this.toReload = false
      },
    },
  }
</script>
