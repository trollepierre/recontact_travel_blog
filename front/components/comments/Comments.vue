<template>
  <section>
    <comment-list
      :to-reload="toReload"
      @reloaded="reloadOff"/>
    <comment-form @reload="reloadTo"/>
  </section>
</template>
<script>
  import CommentForm from './CommentForm.vue'
  import CommentList from './CommentList.vue'

  export default {
    name: 'Comments',
    components: { CommentForm, CommentList },
    data() {
      return {
        toReload: false,
      }
    },
    methods: {
      reloadTo() {
        this.toReload = true
      },
      reloadOff() {
        this.toReload = false
      },
    },
  }
</script>
