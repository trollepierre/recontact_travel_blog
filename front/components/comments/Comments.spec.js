import Comments from '@/components/comments/Comments'
import CommentList from '@/components/comments/CommentList'
import CommentForm from '@/components/comments/CommentForm'

describe('Comments', () => {
  it('should match snapshot', () => {
    // When
    const wrapper = shallowMount(Comments)

    // Then
    expect(wrapper).toMatchSnapshot()
  })

  it('should send false to reload by default', () => {
    // When
    const wrapper = shallowMount(Comments)

    // Then
    expect(wrapper.findComponent(CommentList).props().toReload).toEqual(false)
  })

  it('should send to reload when form asks for reload', async () => {
    // Given
    const wrapper = shallowMount(Comments, { data: () => ({ toReload: false }) })

    // When
    await wrapper.findComponent(CommentForm).vm.$emit('reload')

    // Then
    expect(wrapper.findComponent(CommentList).props().toReload).toEqual(true)
  })

  it('should send to reload off when list returns reloaded', async () => {
    // Given
    const wrapper = shallowMount(Comments, { data: () => ({ toReload: true }) })

    // When
    await wrapper.findComponent(CommentList).vm.$emit('reloaded')

    // Then
    expect(wrapper.findComponent(CommentList).props().toReload).toEqual(false)
  })
})
