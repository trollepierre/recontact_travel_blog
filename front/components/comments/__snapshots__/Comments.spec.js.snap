// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should match snapshot 1`] = `
<section>
  <comment-list-stub></comment-list-stub>
  <comment-form-stub></comment-form-stub>
</section>
`;
